cloud/wfm-rest-api/src/impl/ApiController.ts

```diff
@@ -19,113 +20,98 @@ export class ApiController<T> {
    * Handler for list method
    * Can be reused by developers that wish to mount handler directly on router
    */
-  public listHandler(req: express.Request, res: express.Response) {
-    const self = this;
+  public listHandler(req: Request, res: Response, next: NextFunction) {
     getLogger().debug('Api list method called',
-      { object: self.apiPrefix, body: req.query });
+      { object: this.apiPrefix, body: req.query });
     const page = defaultPaginationEngine.buildRequestFromQuery(req.query);
     let filter = {};
-    if (req.query.filter) {
+    if (req.body.filter) {
+      filter = req.body.filter;
+    } else if (req.query.filter) {
       try {
         filter = JSON.parse(req.query.filter);
       } catch (err) {
-        getLogger().debug('Invalid filter passed');
-        const error: ApiError = new ApiError(errorCodes.CLIENT_ERROR, 'Invalid filter query parameter');
-        return res.status(400).json(error);
+        getLogger().error('Invalid filter passed');
```

Comment:
IMHO rule of thumb is to never log any user specific validation errors as this may be used to DDOS server. That's why I put there debug, but this is so minor message that it should be ok.